bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/camera/CameraHandler.java

```diff
@@ -0,0 +1,144 @@
+            case CHANNEL_CAMERA_SDSTATUS:
+                return getSdStatusState();
+            case CHANNEL_CAMERA_ALIMSTATUS:
+                return getAlimStatusState();
+            case CHANNEL_CAMERA_ISLOCAL:
+                return getIsLocalState();
+            case CHANNEL_CAMERA_LIVEPICTURE_URL:
+                return getLivePictureURLState();
+            case CHANNEL_CAMERA_LIVEPICTURE:
+                return getLivePictureState();
+            case CHANNEL_CAMERA_LIVESTREAM_URL:
+                return getLiveStreamState();
+        }
+        return super.getNAThingProperty(channelId);
+    }
+
+    protected State getStatusState() {
+        return module != null ? toOnOffType(module.getStatus()) : UnDefType.UNDEF;
+    }
+
+    protected State getSdStatusState() {
+        return module != null ? toOnOffType(module.getSdStatus()) : UnDefType.UNDEF;
+    }
+
+    protected State getAlimStatusState() {
+        return module != null ? toOnOffType(module.getAlimStatus()) : UnDefType.UNDEF;
+    }
+
+    protected State getIsLocalState() {
+        return module != null ? toOnOffType(module.getIsLocal()) : UnDefType.UNDEF;
+    }
+
+    protected State getLivePictureURLState() {
+        return getLivePictureURL() == null ? UnDefType.UNDEF : toStringType(getLivePictureURL());
+    }
+
+    protected State getLivePictureState() {
+        return getLivePictureURL() == null ? UnDefType.UNDEF : HttpUtil.downloadImage(getLivePictureURL());
+    }
+
+    protected State getLiveStreamState() {
+        return getLiveStreamURL() == null ? UnDefType.UNDEF : new StringType(getLiveStreamURL());
+    }
+
+    /**
+     * Get the url for the live snapshot
+     *
+     * @return Url of the live snapshot
+     */
+    private String getLivePictureURL() {
+        String result = getVpnUrl();
+        if (result != null) {
+            result += LIVE_PICTURE;
+        }
+        return result;
+    }
+
+    /**
+     * Get the url for the live stream depending wether local or not
+     *
+     * @return Url of the live stream
+     */
+    private String getLiveStreamURL() {
+        String result = getVpnUrl();
+        if (result != null) {
+            result += "/live/index";
+            result += isLocal() ? "_local" : "";
+            result += ".m3u8";
+        }
+        return result;
+    }
+
+    @SuppressWarnings("null")
+    private String getVpnUrl() {
+        return (module == null) ? null : module.getVpnUrl();
+    }
+
+    public String getStreamURL(String videoId) {
+        String result = getVpnUrl();
+        if (result != null) {
+            result += "/vod/" + videoId + "/index";
+            result += isLocal() ? "_local" : "";
+            result += ".m3u8";
+        }
+        return result;
+    }
+
+    @SuppressWarnings("null")
+    private boolean isLocal() {
+        return (module == null || module.getIsLocal() == null) ? false : module.getIsLocal().booleanValue();
+    }
+}
```

Comment:
Some of our build tools don't like EOF without a newline. So add a newline here.

Reply:
done